GunShipAlpha/src/Tests.java

```diff
@@ -35,7 +36,53 @@ void testBehemothLineInteraction() {
         Point end = new Point(450,450);
         Line line = new Line(start,end);
         gameState.addBehemoth(behemoth);
-        boolean isBlocked = behemoth.isIntersectingLine(line);`
+        boolean isBlocked = behemoth.isIntersectingLine(line);
         assertTrue(isBlocked);
     }
+    @Test
+    void testBehemothMovement() {
+        GameState gameState = new GameState();
+        Behemoth behemoth = new Behemoth(350, 350, 50, 50);
+        gameState.addBehemoth(behemoth);
+        double initialDistance = Math.sqrt(Math.pow(350 - 50, 2) + Math.pow(350 - 50, 2));
+        behemoth.moveTowardTarget();
+        Rectangle bounds = behemoth.getBounds();
+        double newDistance = Math.sqrt(Math.pow(bounds.x - 50, 2) + Math.pow(bounds.y - 50, 2));
+        assertTrue(newDistance < initialDistance);
+    }
+    @Test
+    void testRemoveEnemy() {
+        GameState gameState = new GameState();
+        Behemoth behemoth = new Behemoth(50, 50, 400, 400);
+        behemoth.removeHealth();
+        behemoth.removeHealth();
+        gameState.addBehemoth(behemoth);
+        gameState.removeEnemy();
+        assertEquals(0, gameState.getBehemothCount());
+    }
+    @Test
+    void testAddLine() {
+        GameState gameState = new GameState();
+        Point start = new Point(350,350);
+        Point end = new Point(450,450);
+        Component source = new Button();
+        MouseEvent e = new MouseEvent(source, MouseEvent.MOUSE_PRESSED, System.currentTimeMillis(), 0, end.x, end.y, 1, false);
+        gameState.addLine(e, start);
+        assertEquals(1, gameState.getLineListSize());
+    }
+    @Test
+    void testRemovePlayerHealth() {
+        Player player = new Player();
+        for(int i = 0; i < 10; i++) {
+            player.reduceHealth();
+        }
```

Comment:
i like this test. i did wonder if you could have an enemy hitting the player over and over, though, possibly as another test.

Reply:
the enemy can just sit and hit the player like 7 times before it gets deleted its a but I need to work on